From d3cb8f6222a3cb955712b720d7c0c0dba37898f9 Mon Sep 17 00:00:00 2001
From: Guillaume Gardet <guillaume.gardet@arm.com>
Date: Mon, 14 Dec 2020 15:38:22 +0000
Subject: [PATCH 036/147] aarch64: fix static PIE start code for BTI [BZ
 #27068]

A bti c was missing from rcrt1.o which made all -static-pie
binaries fail at program startup on BTI enabled systems.

Fixes bug 27068.

(cherry picked from commit d4136903a29baabeec8987b53081def8b4a49826)
---
 sysdeps/aarch64/start.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysdeps/aarch64/start.S b/sysdeps/aarch64/start.S
index 108f602721b42cac527a0be4f7e2733aec1dc59d..1998ea95d407697fe1b3ced767928092179e4541 100644
--- a/sysdeps/aarch64/start.S
+++ b/sysdeps/aarch64/start.S
@@ -99,6 +99,7 @@ ENTRY(_start)
 	   because crt1.o and rcrt1.o share code and the later must avoid the
 	   use of GOT relocations before __libc_start_main is called.  */
 __wrap_main:
+	BTI_C
 	b	main
 #endif
 END(_start)
-- 
2.35.6

